### ISSUE TITLE
Auto-width for columns and auto-fill to container / window width + height

### ISSUE THREAD

**Issue by GoodMorgon** [Swechhya/excelR#27]
Hello!

Many thanks for developing this package! :) This is absolutely what is needed for developing a shiny app with Excel-based worksheets integrated. 

May I ask if it's on the to-do list to add the features auto-width on column names and another feature to force the width and height to the container width/height? These additions would make the package a killer...

This could best be done in the `excelTable` arguments

Something like this: 

`excelTable(data = mydata, autoWidth = TRUE, autoFill = TRUE)` together with `excelOutput(id = myid, width = "100%", width = "50vh")` should produce auto-spaced columns that fill the whole width and half the height of the screen. 

Instead of having to specify the column width each separately through the `columns` parameter in the `excelTable` function...

**Comment by Swechhya** [comment]
Thank you for pointing out a much needed feature. I will try to include in the future version of the package.

**Comment by marciz** [comment]
In Shiny you can add tags in ui to get auto-width for columns and wrap colnames
But agree would be better to have params in function
```
library(shiny)
library(excelR)
shinyApp(
  ui = fluidPage(
    excelOutput("table1"),
    tags$head(
      tags$style(HTML(".jexcel {
                        width: auto;
                        height: auto;
                        white-space: normal;
                      }"))
    )
  ),
  server = function(input, output, session) {
    output$table1 <- renderExcel({
      dt <- head(iris)
      dt$Species <- paste(dt$Species, dt$Species, dt$Species)
      colnames(dt) <- gsub('\\.', ' ', colnames(dt))
      excelTable(data = dt)
    })
  }
)
```

**Comment by kohlkopf** [comment]
I also have a workaround, for use in an RMarkdown. Wrap your chunk in the following HTML:


`<div class="pagedtable pagedtable-not-empty" style="opacity: 1;"><table style="visibility: hidden; position: absolute; white-space: nowrap; height: auto; width: auto;">`

` ```{r}`
`excelTable(sample_locations, defaultColWidth=200, editable=FALSE)`
` ``` `

`</table></div>`

This results in an auto resizing table. If the output expand past the boundary, you can horizontally scroll to see everything.

**Comment by Swechhya** [comment]
@GoodMorgon , closing this as this issue has been addressed in PR #52. Two params `autoWidth` and `autoFill` has been added to `excelTable` function. Please note that this is currently available in development version only.

Thanks to @marciz for  specifying the style that could be added to get the desired output.
